src/util/tr.js

```diff
@@ -496,75 +498,151 @@ const _tr = (msg, {
+            const heightAdjust = document.getElementById(divId)
+            heightAdjust.style.margin = '0px'
+            heightAdjust.style.height = `${event.data.value}px`
+          }
+          if (event?.data?.action === 'getnotif' + displayObj.layout) {
+            window.clevertap.renderNotificationClicked(event.data.value)
+          }
+
+          if (event?.data?.action === 'Event') {
+            window.clevertap.event.push(event.data.value)
+          } else if (event?.data?.action === 'Profile') {
+            window.clevertap.profile.push(event.data.value)
+          } else if (event?.data?.action === 'OUL') {
+            window.clevertap.onUserLogin.push(event.data.value)
+          } else if (event?.data?.action === 'closeBoxPopUp') {
+            var boxWrapper = window.document.getElementById('wizParDiv0')
+            setTimeout(() => { boxWrapper.remove() }, 0)
+          } else if (event?.data?.action === 'closeBannerPopUp') {
+            var bannerWrapper = window.document.getElementById('wizParDiv2')
+            setTimeout(() => { bannerWrapper.remove() }, 0)
+          }
+        })
+        contentDiv = ''
+      } else {
+        // adjust iframe and body height of html inside correctly
+        contentHeight = document.getElementById('wiz-iframe').contentDocument.getElementById('contentDiv').scrollHeight
+        if (displayObj['custom-editor'] !== true && !isBanner) {
+          contentHeight += 25
+        }
+        document.getElementById('wiz-iframe').contentDocument.body.style.margin = '0px'
+        document.getElementById('wiz-iframe').style.height = contentHeight + 'px'
 
-    const adjustIFrameHeight = () => {
-      // adjust iframe and body height of html inside correctly
-      contentHeight = document.getElementById('wiz-iframe').contentDocument.getElementById('contentDiv').scrollHeight
-      if (displayObj['custom-editor'] !== true && !isBanner) {
-        contentHeight += 25
+        contentDiv = document.getElementById('wiz-iframe').contentDocument.getElementById('contentDiv')
       }
-      document.getElementById('wiz-iframe').contentDocument.body.style.margin = '0px'
-      document.getElementById('wiz-iframe').style.height = contentHeight + 'px'
+      setupClickUrl(onClick, targetingMsgJson, contentDiv, divId, legacy)
     }
 
-    const ua = navigator.userAgent.toLowerCase()
-    if (ua.indexOf('safari') !== -1) {
-      if (ua.indexOf('chrome') > -1) {
-        iframe.onload = () => {
-          adjustIFrameHeight()
-          const contentDiv = document.getElementById('wiz-iframe').contentDocument.getElementById('contentDiv')
-          setupClickUrl(onClick, targetingMsgJson, contentDiv, divId, legacy)
+    iframe.onload = handleIframeLoad
+  }
+
+  const ctEventhandler = (html) => {
+    const ctScript = `
+     var clevertap = {
+      event: {
+        push: (eventName) => {
+          window.parent.postMessage({
+            action: 'Event',
+            value: eventName
```

Comment:
can we have this script in a different file? 
Will reduce the number of lines in this file